### PR TITLE
[INFINITY-1152] Move PlanUtils methods.

### PR DESCRIPTION
**Pull request by mgummelt**

### CODE REVIEW

**Comment by mgummelt** [comment]
@nickbp PTAL

File: sdk/scheduler/src/test/java/com/mesosphere/sdk/scheduler/plan/strategy/ParallelStrategyTest.java [hunk 1]
```diff
@@ -55,45 +60,16 @@ public void testParallelExecution() {
...
-        Assert.assertEquals(1, getCandidates().size());
-        Assert.assertEquals(el2, getCandidates().iterator().next());
-
-        when(el2.isComplete()).thenReturn(true);
-        when(el2.isPending()).thenReturn(false);
-        Assert.assertTrue(getCandidates().isEmpty());
-    }
-
-    @Test
-    public void testDirtyAssetAvoidance() {
```
Comment (nickbp) [inline sdk/scheduler/src/test/java/com/mesosphere/sdk/scheduler/plan/strategy/ParallelStrategyTest.java]: Any reason for this test being dropped? (And a similar one in SerialStrategyTest)

Comment (mgummelt) [inline sdk/scheduler/src/test/java/com/mesosphere/sdk/scheduler/plan/strategy/ParallelStrategyTest.java]: The dirtiness logic is in the Step now, which is mocked out.  So it doesn't belong here, but I suppose I should add the test back into StepTest.

Comment (mgummelt) [inline sdk/scheduler/src/test/java/com/mesosphere/sdk/scheduler/plan/strategy/ParallelStrategyTest.java]: It's now added to `DeploymentStepTest`

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/ParentElement.java [hunk 2]
```diff
@@ -42,4 +51,106 @@ default void updateParameters(Map<String, String> parameters) {
...
+        errors.addAll(parentErrors);
+        Collection<? extends Element> children = getChildren();
+        children.forEach(element -> errors.addAll(element.getErrors()));
+        return errors;
+    }
+
+    @Override
+    default Status getStatus() {
+        // Ordering matters throughout this method.  Modify with care.
+        // Also note that this function MUST NOT call parent.getStatus() as that creates a circular call.
```
Comment (nickbp) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/ParentElement.java]: This second note (on L102) can go away. The equivalent under the new system would be `getStatus()` calling `this.getStatus()` -- the result there shouldn't be surprising for anyone.

**Pull request merged**